Composer/packages/client/src/components/Notifications/useSurveyNotification.tsx

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { useEffect } from 'react';
+import { useRecoilValue } from 'recoil';
+import formatMessage from 'format-message';
+
+import { ClientStorage } from '../../utils/storage';
+import { surveyEligibilityState, dispatcherState, machineInfoState } from '../../recoilModel/atoms/appState';
+import { MachineInfo } from '../../recoilModel/types';
+import { SURVEY_URL_BASE } from '../../constants';
+import TelemetryClient from '../../telemetry/TelemetryClient';
+
+const buildUrl = (info: MachineInfo) => {
+  // User OS
+  // hashed machineId
+  // composer version
+  // maybe include subscription ID; wait for global sign-in feature
+  // session ID  (global telemetry GUID)
+  const version = process.env.COMPOSER_VERSION;
+
+  const parameters = {
+    Source: 'Composer',
+    os: info?.os || 'Unknown',
+    machineId: info?.id,
+    version,
+  };
+
+  return (
+    `${SURVEY_URL_BASE}?` +
+    Object.keys(parameters)
```

Comment:
nit: can use `query-string` stringify method

Reply:
Good idea. I'll fix that.